autogen/main/cluster.tf.tmpl

```diff
@@ -95,6 +95,15 @@ resource "google_container_cluster" "primary" {
       enable_components = var.monitoring_enabled_components
     }
   }
+  dynamic "monitoring_config" {
```

Comment:
There is an existing dynamic monitoring_config, I think we can nest `managed_prometheus` as a dynamic under that.

Reply:
Thanks for quick response and the review comment @bharathkkb! I kept it separate to have a separate `for_each` check as in one case we have `var.monitoring_enabled_components=[]` but we still want managed prometheus enabled. I didn't think of nesting with another dynamic! I will try that!

Reply:
Interestingly switching to a static block:

https://github.com/abousias/terraform-google-kubernetes-engine/blob/c5170bb58cab6706377baf24e58cadc28e685205/autogen/main/cluster.tf.tmpl#L91-L100

It gives an api error:

![image](https://user-images.githubusercontent.com/11448520/175072892-5661c8cc-92ac-44b1-b0af-867083fb1831.png)

Reply:
I am having the error, with the reverted code too. The issue didn't appear because in my test I was applying the change to an existing cluster. Now I am getting this with a brand new cluster.

https://github.com/abousias/terraform-google-kubernetes-engine/blob/c5170bb58cab6706377baf24e58cadc28e685205/autogen/main/cluster.tf.tmpl#L81

I am looking into it.

Reply:
Hey @abousias I have also been working on this. I was able to fix that error by adding a check for the enable managed prometheus variable in the main.tf file.

```logmon_config_is_set      = length(var.logging_enabled_components) > 0 || length(var.monitoring_enabled_components) > 0 || var.monitoring_enable_managed_prometheus```

I was also able to get it working with nested dynamic blocks. Feel free to have a look at what I did if it helps.
https://github.com/jroiseux/terraform-google-kubernetes-engine/tree/managed-prometheus

Reply:
Hey @jroiseux I was about to check exactly the same conditional 😄! I will take a look and copy! Many thanks for the help! Nice to know that it will work 😉!

Reply:
@bharathkkb ready for re-review. It works fine now.

Reply:
⚠️ I discovered that the current version is trying to disable `SYSTEM_COMPONENTS`, here is the plan against an existing cluster:

![image](https://user-images.githubusercontent.com/11448520/175287348-a0320e08-c82f-4527-b5d1-262fbb0d4cb7.png)

Whereas the separate block one leaves the `enabled_components` untouched:

![image](https://user-images.githubusercontent.com/11448520/175287857-15e06eb5-6a3a-459d-a4cd-c95af083914c.png)

This is because the `dynamic` is "forced" to produce the block but the `var.monitoring_enabled_components` is `[]`. My existing cluster is also based on the main module instead of the `beta` ones. I am checking if that is an issue for me.

@jroiseux this is most probably happening with your version, isn't it?

Reply:
@abousias Yes, it happens on my version too.

This is what the statefile looks like when I use the current module for "beta-private-cluster" (without the changes from this PR) without setting `var.monitoring_enabled_components` :

![Screenshot from 2022-06-23 09-16-34](https://user-images.githubusercontent.com/39174028/175346940-4eed6286-349d-4a7d-b239-0b56f3c456a9.png)

I think it might be an issue with the default value. As `enabled_components` seems to end up as `["SYSTEM_COMPONENTS"]`, even though the default is `[]`.

Reply:
@bharathkkb I am reverting back to the original implementation, leaving TF to merge the blocks in one well after the evaluation of the the `for_each`s.  As @jroiseux also points out, the other option would be to change the **default** value of `var.monitoring_enabled_components` to something like `[ "SYSTEM_COMPONENTS" ]` which I don't know if it is a good idea.